### ISSUE TITLE
Document TextSpan::locale in accessibility docs

### ISSUE THREAD

**Issue by Leffe108** [flutter/flutter#99596]
## Use case
In order to markup which voice locale that assistive technology like TalkBack and VoiceOver should use, a Flutter user/developer has to use `TextSpan::locale`. This is however neither mentioned in [accessibility doc](https://docs.flutter.dev/development/accessibility-and-localization/accessibility) nor [internationalization doc](https://docs.flutter.dev/development/accessibility-and-localization/internationalization)

It could also help to clear up that the locale property on MaterialApp and Localizations.override does not affect which voice assistive technologies like VoiceOver and TalkBack is using as opposed to how the lang attribute works in HTML. [Reference](https://github.com/flutter/flutter/issues/99018#issuecomment-1057497573)

## Proposal

At the end of the [Screen reader section](https://docs.flutter.dev/development/accessibility-and-localization/accessibility#screen-readers), add a paragraph

> When there is text is your app that should be voiced with a specific voice, use [`TextSpan.locale`](https://api.flutter.dev/flutter/painting/TextSpan/locale.html) to inform the screen reader about the voice to use. Note that `MaterialApp.locale` and `Localizations.override` do not affect which voice is used by the screen reader. Usually the screen reader will use the system voice to voice texts except where you explicitly told it with `TextSpan.locale`.

**Comment by Leffe108** [comment]
As a comment, using eg. Swedish voice to read English text gives poor audibility compared to if the assistive tech is using English voice for English content. As many people are bi- or multi lingual and consume apps and content in many languages, I think this is an important note to the docs. - Both documenting the facilities that does exist in Flutter and also documenting the "false friend" that Flutter is not sending the locale of the app to assistive tech like HTML/browsers does.
